### PR TITLE
added memoized fibonacci

### PR DESCRIPTION
**Pull request by Bhupesh-V**
**Fixes issue:**

**Changes:**

- Fibonacci Series using Memoization

### CODE REVIEW

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* Fibonacci Series implemented using Memoization */
+#include <iostream>
+using namespace std;
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Don't use this.

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+long long fibonacci (int n) {
+
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Extra newline.

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    static long long fib[100] = {}; // initialises the array with all elements as 0
+    fib[1] = 0; fib[2] = 1;
+	if (n == 1 || n == 2) return fib[n];
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Fix indentation in this function.

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+int main () {
+	
+	int n;
+	cout << "Enter number of terms: ";
+	cin >> n;
+	for (int i = 1; i <= n; i++) 
+		cout << fibonacci (i) << endl; 	
+
+	return 0;
+}
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Add a newline.

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+int main () {
+	
+	int n;
+	cout << "Enter number of terms: ";
+	cin >> n;
+	for (int i = 1; i <= n; i++) 
+		cout << fibonacci (i) << endl; 	
+
+	return 0;
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]:
```suggestion

```

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp (hunk 1 above)
Comment (Bhupesh-V) [reply code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Hey so should i use `std::cout` etc

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp (hunk 4 above)
Comment (Bhupesh-V) [reply code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: I am not able to process what are you saying
should i add a new line after `return 0;` statement ?

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp (hunk 1 above)
Comment (Bhupesh-V) [reply code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: done

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp (hunk 2 above)
Comment (Bhupesh-V) [reply code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: done

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp (hunk 4 above)
Comment (arnavb) [reply code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: @Bhupesh-V I meant to add a newline at the end of the file.

File: code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/* Fibonacci Series implemented using Memoization */
+#include <iostream>
+
+long long fibonacci (int n) {
+    static long long fib[100] = {}; // initialises the array with all elements as 0
+    fib[1] = 0; 
+    fib[2] = 1;
+	
+	if (n == 1 || n == 2)
```
Comment (arnavb) [inline code/unclassified/src/optimized_fibonacci/optimized_fibonacci.cpp]: Indentation still needs to be fixed.

**Comment by arnavb** [comment]
djldk 2

**Pull request merged**